src/utils/loggable.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import os
+
+
+DEFAULT_LOG_LEVEL = logging.WARNING
+
+_COLOR_TMPL = "\033[3{}m"
+(RED,
+ GREEN,
+ YELLOW,
+ BLUE,
+ VIOLET,
+ CYAN) = map(lambda x: _COLOR_TMPL.format(x), range(1, 7))
+
+BOLD = "\033[1m"
+ESC = "\033[0m"
+
+
+def apply_style(text, color=None, bold=False):
+    bold = BOLD if bold else ""
+    color = color or ""
+    return "{}{}{}{}".format(bold, color, text, ESC)
+
+
+def get_log_level():
+    level = os.environ.get('LOGLEVEL')
+    if not level:
+        return DEFAULT_LOG_LEVEL
+    try:
+        level = int(level)
+    except ValueError:
+        level = logging.__dict__.get(level)
+    if isinstance(level, int):
+        return level
+    return DEFAULT_LOG_LEVEL
+
+
+class DefaultFormatter(logging.Formatter):
+    _DEFAULT_FORMAT = ("%(levelname)s : %(asctime)s %(funcName)s"
+                       " - %(message)s (%(filename)s:%(lineno)d)")
+    _DEFAULT_COLORS = {
+        "CRITICAL": VIOLET,
+        "ERROR": RED,
+        "WARNING": YELLOW,
+        "INFO": CYAN,
+        "DEBUG": BLUE
+    }
+
+    def __init__(self, obj, beautify=True):
+        super().__init__(self._DEFAULT_FORMAT)
+        self.obj = obj
+        self.beautify = beautify
+
+    def format(self, record):
+        if self.beautify:
+            color = self._DEFAULT_COLORS[record.levelname]
+            record.levelname = apply_style(record.levelname, color, bold=True)
+            record.levelname = record.levelname.ljust(21)
+        else:
+            record.levelname = record.levelname.ljust(8)
+        return super().format(record)
+
+
+class ObjectFormatter(DefaultFormatter):
+    _DEFAULT_FORMAT = ("%(levelname)s : %(asctime)s %(name)s.%(funcName)s"
+                       " - %(message)s (%(filename)s:%(lineno)d)")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def format(self, record):
+        record.name = "<{} at {}>".format(self.obj.__class__.__name__,
```

Comment:
If I look closer at the logs I think we don't need this. It's only printing `<HackSoundServer at 140020789004832>` and there is only one hack sound server per process anyway.

Reply:
That's right. I agree.